### ISSUE TITLE
Allow to open container in new workspace with for_window

### ISSUE THREAD

**Issue by marcosnils** [i3/i3#3732]
## I'm submitting a…

<pre>
[ ] Bug
[x] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior
First of all apologies if this is not the place to ask this question. I've tried to reach someone in IRC but even though the channel is filled, I didn't get any answers and it doesn't have archive to check if someone eventually responded.

What I'm trying to do is to use `for_window` to open the selected window in a new empty workspace. Reading through the docs, the only options are `move, next, prev, etc` but I couldn't find a way to send the container to a new pristine workspace.

Is such feature available?

Thx in advance.

**Comment by Airblader** [comment]
You can use for_window to mark the window or send it to some special workspace, then write a script using the IPC that listens for the mark / workspace, figures out which (new) workspace to send it to and do it.

You can also cut out the for_window part and have the script do everything if for example you want to do it for all windows anyway.

**Comment by marcosnils** [comment]
thx @Airblader how about suggesting a feature to make for_window do such action by default?
i.e `for_window for_window [class="^evil-app$"] floating enable, move container to workspace new`
this is exactly what I'm looking for.

If you think it might be a good idea, with some guidance I can give it a try..

**Comment by Airblader** [comment]
There are two problems with that:

1. The syntax is ambiguous. It already works, but it would move it to a workspace named "new".
2. It's entirely unclear how to name such a new workspace and users would most likely want to use different ways of naming them. This poses a high risk of feature creep. 

I also don't think the usecase here is crazy big, so before investing the complexity of a new feature I'd go with a script and see if the feature request accumulates interest.

**Comment by marcosnils** [comment]
> 1. The syntax is ambiguous. It already works, but it would move it to a workspace named "new".

right. Isn't that the same with the `prev, next`? Those are special cases, so if you name a workspace `next or prev` it wouldn't move it to that workspace but instead use the built-in support for that.



> 2\. It's entirely unclear how to name such a new workspace and users would most likely want to use different ways of naming them. This poses a high risk of feature creep.

Agree. Maybe assign a number? Just doing some quick brainstorming here.



> I also don't think the usecase here is crazy big, so before investing the complexity of a new feature I'd go with a script and see if the feature request accumulates interest.

Yep, that's correct. I just wanted to know if there was some built-in support for that. I guess I'll just script it out and we'll see if it gets any attention. Thx for your input and taking the time to respond.

**Comment by Airblader** [comment]
> Isn't that the same with the prev, next?

Technically, yes, you're right. :-) But those are already there, while adding a new reserved word could in theory break for users (for example if they have such a script already). Using something like `--new` instead would easily solve this minor issue, though.

**Comment by Airblader** [comment]
(Pressed Enter too quickly)

> Maybe assign a number?

Yes, but my concern is that we'd likely quickly have users wanting different things here and if the feature cannot even be used by all those who want to use it it becomes even less useful.

**Comment by sflip** [comment]
I just want to give a hint: I personally use [quickswitch](https://github.com/OliverUv/quickswitch-for-i3) which offers additional features for i3, e.g. to move an existing container to a new workspace: `quickswitch --journey -E`.

**Comment by asoymos** [comment]
Hi (sorry for lenguage mistakes, english is second to me). I post this here as it's still open.
Everything that I had imagine I can do with i3 (thank you!), except this. I have found many people interested in the feature [(one comment](https://www.reddit.com/r/i3wm/comments/132jcpc/anyone_know_how_to_move_container_to_the_next/) would even say that someone mentioned this everyday). Just in this repository I counted 5-6 issues when I though was enough. [Here](https://github.com/i3/i3/issues/5484) a i3 member proposed a script, in fact there are many scripts for doing this that work with specific setups. Like [1](https://github.com/v1nns/dotfiles/blob/main/.local/bin/new-workspace.py), [2](https://github.com/altdesktop/i3ipc-python/blob/master/examples/workspace-new.py), [3](https://github.com/regolith-linux/i3-next-workspace), [4](https://github.com/roguh/i3empty). So I think that to be able to switch to a new empty workspace without needing to check which one is avaliable (like next_new or something) is a no brainer. Just wanted to contribute my vote. Thanks.
